pkg/sunmao/sdk.go

```diff
@@ -10,78 +10,170 @@ import (
 
 // App
 
+type Builder interface {
+	ValueOf() any
+}
+
+type BaseBuilder[T Builder] interface {
```

Comment:
Does this mean the ModuleBuilder also has `NewTrait` and `Component` methods in its interface? Which I think it does not have in sunmao-ui.

Reply:
Yes. But i found that module's `impl` is the same meaning with `component` in Application actually (am i miss understanding ?).

Reply:
Yep, and that's why I think we can make the module's interface much simpler than the app builder.

For example:

```go
m := sunmao.NewModule().Type("custom/v1/file_type").
  Impl(b.newText()) // <- just re-use the app builder to get component schema
```

With this API, we do not need to implement module's `newComponent` and `newTrait` interfaces since it has no difference from the app builder's.